endorsement/models/shared_drive.py

```diff
@@ -42,97 +59,108 @@ def __str__(self):
 
 
 class SharedDriveMember(
-        ExportModelOperationsMixin('shared_drive_member'), models.Model):
+    ExportModelOperationsMixin("shared_drive_member"), models.Model
+):
     """
     Member model represents users/groups responsible the given
     shared drive and corresponding subscription.
     """
+
     member = models.ForeignKey(Member, on_delete=models.PROTECT)
     role = models.ForeignKey(Role, on_delete=models.PROTECT)
 
     def json_data(self):
         return {
             "name": self.member.json_data(),
-            "role": self.role.json_data()
-            }
+            "role": self.role.json_data(),
+        }
 
     def __str__(self):
         return json.dumps(self.json_data())
 
 
 class SharedDriveQuota(
-        ExportModelOperationsMixin('shared_drive_tier'), models.Model):
+    ExportModelOperationsMixin("shared_drive_tier"), models.Model
+):
     """
     SharedDriveQuota model represents a quota (tier)
 
     Quota limit is represnted as an integer number of Gigabytes
     """
+
     org_unit_id = models.CharField(max_length=32)
-    org_unit_name = models.CharField(max_length=64, null=True)
+    org_unit_name = models.CharField(
+        max_length=64, null=True
+    )  # TODO: shouldn't this be non-NULL?
     quota_limit = models.IntegerField(null=True)
 
     @property
     def is_subsidized(self):
         return self.quota_limit <= getattr(
-            settings, 'ITBILL_SHARED_DRIVE_SUBSIDIZED_QUOTA')
+            settings, "ITBILL_SHARED_DRIVE_SUBSIDIZED_QUOTA"
+        )
 
     def json_data(self):
         return {
             "org_unit_id": self.org_unit_id,
             "org_unit_name": self.org_unit_name,
             "quota_limit": self.quota_limit,
-            "is_subsidized": self.is_subsidized
+            "is_subsidized": self.is_subsidized,
         }
 
 
-class SharedDrive(ExportModelOperationsMixin('shared_drive'), models.Model):
+class SharedDrive(ExportModelOperationsMixin("shared_drive"), models.Model):
     """
     SharedDrive model represents a shared drive, its current quota
     and its members
     """
-    drive_id = models.SlugField(max_length=32)
+
+    # NOTE: maximum observed is 19 characters (2024-05-16)
+    drive_id = models.SlugField(max_length=25)
```

Comment:
Reduced from 32 => 25 so that the new method `get_itbill_key_remote()` would never return a value <= length rquired by ITBillSubscription